File: April 2020/11 - Diameter of Binary Tree/Solution.cpp
class Solution {
    int ans;
public:
    int diameterOfBinaryTree(TreeNode* root) {
        ans = 1; // root node
        height(root);
        return ans - 1;
    }
    int height(TreeNode* node) {
        if(node == NULL) return 0;
        int Left = height(node->left);
        int Right = height(node->right);
        ans = max(ans, Left + Right + 1); // compare diameter
        return max(Left, Right) + 1; // height of a subtree     
    }
};

File: April 2020/11 - Diameter of Binary Tree/Solution.py
class Solution(object):
    def diameterOfBinaryTree(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        self.ans = 1
        def depth(node):
            if node is None:
                return 0
            r = depth(node.right)
            l = depth(node.left)
            self.ans = max(self.ans, l + r + 1) # compare diameter of subtree/tree
            return max(l, r) + 1 # compare height of subtree/tree
        depth(root)
        return self.ans - 1